File: frontend/src/types/session.ts
import type { Message } from './chat';

export type Industry =
  | 'ecommerce'
  | 'finance'
  | 'healthcare'
  | 'education'
  | 'manufacturing'
  | 'internet'
  | 'hr'
  | 'marketing'
  | 'other';

export interface Session {
  session_id: string;
  created_at: string;
  updated_at: string;
  file_name: string;
  industry?: Industry;
  first_query: string;
  methods_used: string[];
  message_count: number;
}

export interface SessionDetail extends Session {
  messages: Message[];
  data_summary: DataSummary;
  report_conclusion?: string | null;
}

export interface ColumnStats {
  mean: number;
  std: number;
  min: number;
  max: number;
}

export interface DataSummary {
  rows: number;
  columns: number;
  column_names: string[];
  column_types: Record<string, string>;
  column_stats?: Record<string, ColumnStats>;
}

export interface SessionsQuery {
  page?: number;
  size?: number;
  keyword?: string;
  industry?: Industry;
  method?: string;
  start_date?: string;
  end_date?: string;
}

export interface SessionsResponse {
  total: number;
  page: number;
  items: Session[];
}

File: backend/app/schemas/session.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ColumnStats(BaseModel):
    mean: float
    std: float
    min: float
    max: float


class DataSummary(BaseModel):
    rows: int
    columns: int
    column_names: list[str]
    column_types: dict[str, str]
    column_stats: dict[str, ColumnStats] = Field(default_factory=dict)


class Session(BaseModel):
    session_id: str
    created_at: str
    updated_at: str
    file_name: str | None = None
    industry: str | None = None
    first_query: str | None = None
    methods_used: list[str] = Field(default_factory=list)
    message_count: int = 0


class SessionsResponse(BaseModel):
    total: int
    page: int
    items: list[Session]


class SessionsQuery(BaseModel):
    page: int | None = 1
    size: int | None = 10
    keyword: str | None = None
    industry: str | None = None
    method: str | None = None
    start_date: str | None = None
    end_date: str | None = None


class Message(BaseModel):
    id: str
    role: str
    content: str
    timestamp: str
    analysis: dict | None = None


class SessionDetail(Session):
    messages: list[Message]
    data_summary: DataSummary | None = None
    report_conclusion: str | None = None
